src/NATS.Client.ObjectStore/NatsObjStore.cs

```diff
@@ -530,17 +531,33 @@ public async ValueTask<ObjectMetadata> GetInfoAsync(string key, bool showDeleted
         {
             LastBySubj = GetMetaSubject(key),
         };
+
         try
         {
             var response = await _stream.GetAsync(request, cancellationToken);
-
-            if (response.Message.Data == null)
+            if (response.Message.Data.Length == 0)
             {
                 throw new NatsObjException("Can't decode data message value");
             }
 
-            var base64String = Convert.FromBase64String(response.Message.Data);
-            var data = NatsObjJsonSerializer<ObjectMetadata>.Default.Deserialize(new ReadOnlySequence<byte>(base64String)) ?? throw new NatsObjException("Can't deserialize object metadata");
+            var bytes = ArrayPool<byte>.Shared.Rent(response.Message.Data.Length);
+            ObjectMetadata data;
+            try
+            {
+                if (System.Buffers.Text.Base64.DecodeFromUtf8(response.Message.Data.Span, bytes.AsSpan(),out _, out var written) == OperationStatus.Done)
+                {
+                    var buffer = new ReadOnlySequence<byte>(bytes.AsMemory(0, written));
+                    data = NatsObjJsonSerializer<ObjectMetadata>.Default.Deserialize(buffer) ?? throw new NatsObjException("Can't deserialize object metadata");
+                }
+                else
+                {
+                    throw new NatsObjException("Can't decode data message value");
+                }
+            }
+            finally
+            {
+                ArrayPool<byte>.Shared.Return(bytes);
+            }
```

Comment:
Ironically, this optimizes NET8 too 😁. Not sure how much of a hot path this can be but why not?

That said, If there is one regret, IDK how to make this easily 'composable where this doesn't feel like a copypaste of the parse with buffer'. OTOH I had had a colleague spend some time teaching me go so the overton window for pragmatism has changed. 🤷